### PR TITLE
option to hide the publish date 

### PR DESCRIPTION
**Pull request by belljohan3**

### CODE REVIEW

**Review by ivanslabbert — CHANGES_REQUESTED** [review]
Good work @belljohan3 

Some comments:

- The field must also be added to ContentDto, and copied from the incoming PostDto/TagDto in the API
- I think we need to add something to set the default value of `publishDateVisible` to false when creating a new Tag. On Posts the default of true is good. Should be ok to only do this in the CMS.
- Add a test to check if the publish date is shown when `publishDateVisible` is `undefined` (if you have not done so already)
- Add a Add a test to check if the publish date is hidden when `publishDateVisible` is `false` (if you have not done so already)

**Comment by belljohan3** [comment]
As discussed, it might be better to run an upgrade in the database schema with functionnality added to #290.

**Comment by ivanslabbert** [comment]
@belljohan3 we should please go through the conflicts together to avoid reverting recent changes that was merged to main?

**Comment by ivanslabbert** [comment]
@belljohan3 I am getting an error in the CMS console when loading an existing Post / Tag document:
image

File: cms/src/components/content/EditContent.vue [hunk 1]
```diff
@@ -125,6 +128,7 @@ const createTranslation = (language: LanguageDto) => {
         title: `Translation for ${language.name}`,
         slug: "",
         parentTags: [],
+        publishDateVisible: true,
```
Comment (ivanslabbert) [inline cms/src/components/content/EditContent.vue:131]: No need to set the `publishDateVisible` field on the ContentDto documents as this is done in the API.

File: cms/src/components/content/EditContentParentValidation.vue [hunk 2]
```diff
@@ -69,6 +69,7 @@ const createTranslation = (language: LanguageDto) => {
         title: `Translation for ${language.name}`,
         slug: "",
         parentTags: [],
+        publishDateVisible: true,
```
Comment (ivanslabbert) [inline cms/src/components/content/EditContentParentValidation.vue:72]: No need to set the `publishDateVisible` field on the `ContentDto` documents as this is done in the API.

File: cms/src/tests/mockdata.ts [hunk 3]
```diff
@@ -21,6 +21,7 @@ export const mockCategoryDto: TagDto = {
     pinned: false,
     image: "test-image.jpg",
     tags: [],
+    publishDateVisible: true,
```
Comment (ivanslabbert) [inline cms/src/tests/mockdata.ts:24]:
```suggestion
    publishDateVisible: false,
```

File: cms/src/tests/mockdata.ts [hunk 4]
```diff
@@ -32,6 +33,7 @@ export const mockTopicDto: TagDto = {
...
```
Comment (ivanslabbert) [inline cms/src/tests/mockdata.ts:36]:
```suggestion
    publishDateVisible: false,
```

File: cms/src/tests/mockdata.ts [hunk 5]
```diff
@@ -125,6 +131,7 @@ export const mockCategoryContentDto: ContentDto = {
     text: '{"type":"doc","content":[{"type":"paragraph","content":[{"type":"text","text":"A category"}]}]}',
     publishDate: 1704114000000,
     parentImage: "test-image.jpg",
+    publishDateVisible: true,
```
Comment (ivanslabbert) [inline cms/src/tests/mockdata.ts:134]:
```suggestion
    publishDateVisible: false,
```

File: cms/src/tests/mockdata.ts [hunk 6]
```diff
@@ -143,6 +150,7 @@ export const mockTopicContentDto: ContentDto = {
...
```
Comment (ivanslabbert) [inline cms/src/tests/mockdata.ts:153]:
```suggestion
    publishDateVisible: false,
```

File: api/src/dto/ContentDto.ts [hunk 7]
```diff
@@ -98,4 +107,9 @@ export class ContentDto extends _contentBaseDto {
     @IsString()
     @Expose()
     parentTagType?: TagType;
+
+    @IsOptional() // Optional as it is set upon change request processing
+    @IsBoolean()
+    @Expose()
+    publishDateVisible?: boolean;
```
Comment (ivanslabbert) [inline api/src/dto/ContentDto.ts:114]: This is perhaps debatable, but as this field is replicated from the parent post/tag document, it should perhaps be renamed to `parentPublishDateVisible` to keep it in line with other replicated fields?

**Comment by belljohan3** [comment]
Yes I am aware of that. It is coming from EditContentParent component. It seems like we need a computed function that set a the new value to the publishdateVisible when the parent changed

**Review by ivanslabbert — CHANGES_REQUESTED** [review]
As an upgrade script has been added updating existing documents to include `publishDateVisible` field, we can remove code in the CMS and app handling cases where `publishDateVisible` is undefined. The field can also be made mandatory (currently it is optional).

**Comment by ivanslabbert** [comment]
> Yes I am aware of that. It is coming from EditContentParent component. It seems like we need a computed function that set a the new value to the publishdateVisible when the parent changed

Or use a v-if on the toggle component?

**Comment by belljohan3** [comment]
> > Yes I am aware of that. It is coming from EditContentParent component. It seems like we need a computed function that set a the new value to the publishdateVisible when the parent changed
> 
> Or use a v-if on the toggle component?

I think as the field is now mandatory, the warning is not showing anymore.

File: app/src/components/content/ContentTile.spec.ts [hunk 8]
```diff
@@ -60,6 +60,16 @@ describe("ContentTile", () => {
         expect(wrapper.text()).not.toContain("Jan 1, 2024");
     });
 
+    it("hides the publishDate is false", async () => {
```
Comment (ivanslabbert) [inline app/src/components/content/ContentTile.spec.ts:63]:
```suggestion
    it("hides the publishDate of publishDateVisible is false", async () => {
```

File: app/src/components/content/ContentTile.vue [hunk 9]
```diff
@@ -25,7 +25,10 @@ const openContent = () => {
     >
         <img :src="content.parentImage" class="aspect-video rounded-lg object-cover shadow-md" />
         <h3 class="mt-2 text-sm text-zinc-800 dark:text-zinc-50">{{ content.title }}</h3>
-        <div v-if="showPublishDate" class="mt-0.5 text-xs text-zinc-500 dark:text-zinc-200">
+        <div
+            v-if="props.showPublishDate && content.parentPublishDateVisible"
```
Comment (ivanslabbert) [inline app/src/components/content/ContentTile.vue:29]: Not needed to include `props.xxx` in the html as Vue does this automatically

**Review by ivanslabbert — APPROVED** [review]
Approved, but please have a look at two minor things I commented on?

**Pull request merged**